plugins/modules/ip.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+"""This module allows users to allocate a new IPv4 Address on their accounts."""
+
+from __future__ import absolute_import, division, print_function
+
+from typing import Any, Optional
+
+import ansible_collections.linode.cloud.plugins.module_utils.doc_fragments.ip as docs
+from ansible_collections.linode.cloud.plugins.module_utils.linode_common import (
+    LinodeModuleBase,
+)
+from ansible_collections.linode.cloud.plugins.module_utils.linode_docs import (
+    global_authors,
+    global_requirements,
+)
+from ansible_collections.linode.cloud.plugins.module_utils.linode_helper import (
+    filter_null_values,
+)
+from ansible_specdoc.objects import FieldType, SpecDocMeta, SpecField
+
+spec: dict = {
+    "linode_id": SpecField(
+        type=FieldType.integer,
+        required=True,
+        description=[
+            "The ID of a Linode you have access to "
+            "that this address will be allocated to."
+        ],
+    ),
+    "public": SpecField(
+        type=FieldType.bool,
+        required=True,
+        description=["Whether to create a public or private IPv4 address."],
+    ),
+    "type": SpecField(
+        type=FieldType.string,
+        required=True,
+        choices=["ipv4"],
+        description=[
+            "The type of address you are requesting. "
+            "Only IPv4 addresses may be allocated through this operation."
+        ],
+    ),
```

Comment:
Could a `state` attribute be added here to support creating/deleting IPs?

Reply:
I find delete IP is not properly implemented in python sdk. We can't use the delete() function from Base because the endpoint is not correct. I'll create a python-sdk ticket to implement the right endpoint.